sys/vane/jael.hoon

```diff
@@ -624,11 +626,29 @@
     ~(able ~(ex ur urb) who)
   ::                                                    ::  ++scry:of
   ++  scry                                              ::  read
-    |=  {syd/@tas pax/path}  ^-  (unit gilt)
+    |=  {syd/@tas pax/path}
+    ~|  %jael-scry-of-stub
     =^  mar  pax  =/(a (flop pax) [-.a (flop t.+.a)])
-    ?>  ?=(_-:*gilt mar)
-    =-  (biff - (flit |=(a/gilt =(-.a mar))))
-    ~  ::TODO
+    !!
+  ::                                                    ::  ++sein:of
+  ++  sein                                              ::  sponsor
+    |=  who=ship
+    ^-  ship
+    ::  XX save %dawn sponsor in .own.sub, check there
+    ::
+    =/  hul  (~(get by hul.eth.sub) who)
+    ?:  ?&  ?=(^ hul)
+            ?=(^ net.u.hul)
+            ?=(^ sponsor.u.net.u.hul)
+        ==
+      u.sponsor.u.net.u.hul
+    (^sein:title who)
```

Comment:
That is not correct. If the sponsor is unset (but the ship is otherwise active/initialized, ie it has an owner (for our purposes here saying "it has network keys" is also fair, since anything without network keys does not matter to the urbit network)) then that means the ship has no sponsor at all.

Reply:
To be clear, "`+hull` as a representation of the on-chain state" is exactly correct, but the on-chain logic allows for a "no sponsor" state.

Reply:
If the sponsor has not been changed and is still just the prefix, is that captured on-chain? Or is it implicit in that field being unset?

Reply:
The field gets set on ship creation, so all ships start out with the prefix sponsor. If the sponsor changes, the value here changes. If the sponsor kicks, this value becomes `~`.  
There is no implicit state in a `+hull` that has been updated using the `apply-hull-diff` stdlib logic.

Reply:
Right, but is that true of the contract?

Reply:
That is also true of the contract. No implicit values there. The only "no sponsor set" case that isn't strictly "it got kicked" is for unactivated/unspawned/non-existent ships.

Reply:
Ok, thanks. I'll put the initialization back in. And we need to decide what to do with the sein scry in the case of lost sponsorship. I think you had mentioned falling back to the last set sponsor, which will require state structure updates in %jael. We could produce a double unit, but that will crash in userspace and force vanes to handle it.

Is their anything we would want to do in %ames other than continue trying to use the old sponsor for routing?

//cc @cgyarvin